fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/OrElseCursor.java

```diff
@@ -154,4 +214,76 @@ public boolean accept(@Nonnull RecordCursorVisitor visitor) {
         }
         return visitor.visitLeave(this);
     }
+
+    private enum State {
```

Comment:
Given that this is private anyway, is there a reason to not use the proto-generated enum?

Reply:
No, except that it was my understanding that we did not idiomatically use protobuf generated classes except when interfacing directly with protobufs. I'd be more than happy to drop this enum.

Reply:
I think we probably generally want to keep generated proto out of our API...though we maybe have already broken that rule. But directly interfacing with protos in private methods seems fine to me. This also isn't super high priority